.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "tests/ubench/google-benchmark"]
```

Comment:
In this instance, I think that these alternatives would be worse than using a git submodule:
* the source comprises a considerable number of files; incorporating then would add a lot to our tree while (in principle) we would not be updating them with new local revisions;
* ideally the benchmark library and benchmarked code would be built with the sample compiler and options, which would generally not be the case for an externally built component.

The 'standard' CMake approach would be to clone the google benchmark repository at CMake configure time, but this leads to a lot of redundant downloads and access to github not only at download time, but also configure/build time.

On the other hand, most of the issues we had with git submodules in the past arose due to our needing to update the submodules both from without the project and within. For this application, the gooogle benchmark code would correspond to a single revision that would need to be updated rarely, if ever, and we would not be pushing changes back upstream from the submodule.

I propose adding a test to the CMake script that checks if the submodule has been properly instantiated (or rather, test that there is a `.git` subdirectory in the subfolder) and either warns that the submodule needs to be updated or attempts to do so itself.